tests/app/main/test_suppliers.py

```diff
@@ -0,0 +1,216 @@
+import mock
+from mock import Mock
+from nose.tools import assert_equal, assert_true, assert_in, assert_false
+from tests.app.helpers import BaseApplicationTest
+
+
+def get_supplier(*args, **kwargs):
+    return {"suppliers": {
+        "id": 1234,
+        "description": "Supplier Description",
+        "clients": ["Client One", "Client Two"],
+        "contactInformation": [{
+            "id": 2,
+            "website": "supplier.dmdev",
+            "email": "supplier@user.dmdev",
+            "contactName": "Supplier Person",
+            "phoneNumber": "0800123123",
+            "address1": "1 Street",
+            "address2": "2 Building",
+            "city": "Supplierville",
+            "postcode": "11 AB",
+        }]
+    }}
+
+
+class TestSuppliersDashboard(BaseApplicationTest):
+    @mock.patch("app.main.views.suppliers.data_api_client")
+    def test_shows_supplier_info(self, data_api_client):
+        data_api_client.get_supplier.side_effect = get_supplier
+        with self.app.test_client():
+            self.login()
+
+            res = self.client.get("/suppliers")
+            assert_equal(res.status_code, 200)
+
+            data_api_client.get_supplier.assert_called_once_with(1234)
+
+            resp_data = res.get_data(as_text=True)
+
+            assert_in("Supplier Description", resp_data)
+            assert_in("Client One", resp_data)
+            assert_in("Client Two", resp_data)
+
+            assert_in("1 Street", resp_data)
+            assert_in("2 Building", resp_data)
+            assert_in("supplier.dmdev", resp_data)
+            assert_in("supplier@user.dmdev", resp_data)
+            assert_in("Supplier Person", resp_data)
+            assert_in("0800123123", resp_data)
+            assert_in("Supplierville", resp_data)
+            assert_in("11 AB", resp_data)
+
+    @mock.patch("app.main.views.suppliers.data_api_client")
+    def test_shows_edit_buttons(self, data_api_client):
+        data_api_client.get_supplier.side_effect = get_supplier
+        with self.app.test_client():
+            self.login()
+
+            res = self.client.get("/suppliers")
+            assert_equal(res.status_code, 200)
+
+            assert_true("/suppliers/edit" in res.get_data(as_text=True))
```

Comment:
This could test that these are the href values for a particular link - just having these strings in the response doesn't really guarantee that the edit buttons exist.